### ISSUE TITLE
Could someone explain the output in an example?

### ISSUE THREAD

**Issue by ykasimov** [tensorflow/swift#161]
There is an example in [here](https://github.com/tensorflow/swift/blob/master/docs/DifferentiableFunctions.md)

```
struct Vector: Differentiable, VectorNumeric {
    var x, y: Float

    // Differentiable computed property.
    @differentiable // Implicitly: @differentiable(wrt: self)
    var magnitude: Float {
        return (x * x + y * y).squareRoot()
    }

    // Differentiable initializer.
    @differentiable // Implicitly: @differentiable(wrt: (x, y))
    init(x: Float, y: Float) {
        self.x = x
        self.y = y
    }
}

let v = Vector(x: 2, y: 2)
print(v.magnitude)
// 2.828427
print(gradient(at: v) { v in v.magnitude })
// Vector(x: 64.0, y: 64.0)
```

I am not sure this output is correct. Because magnitude is calculated like this: (x^2+y^2)^0.5
and the gradient is ( x/(x^2+y^2)^0.5 , y/(x^2+y^2)^0.5 ) and for x=2, y=2 the result is (2/2.828427). and not 64.0.
Also, it seems the output is correct for the function (x^2+y^2)^2
The gradient produced for this function seems to be the same as for the square root. 

I might not understand correctly what is being calculated. 
Could someone please explain it to me?

**Comment by dan-zheng** [comment]
Hi @ykasimov,

Thanks for the catch! `2/2.828427` is indeed the correct gradient value.
`squareRoot` had an incorrect derivative that was [fixed after the v0.2 release](https://github.com/apple/swift/pull/23139).

**Comment by dan-zheng** [comment]
Correct gradient value updated in https://github.com/tensorflow/swift/pull/162.
